From 35a39d6953aff5896f6d516dba2e5bb26913781c Mon Sep 17 00:00:00 2001
From: MisileLab <misileminecord@gmail.com>
Date: Sat, 18 Nov 2023 20:17:25 +0900
Subject: [PATCH] a flatpak thing

Signed-off-by: MisileLab <misileminecord@gmail.com>
---
 bdsx.sh | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/bdsx.sh b/bdsx.sh
index 74eff543..dffd053c 100755
--- a/bdsx.sh
+++ b/bdsx.sh
@@ -9,8 +9,14 @@ if [ -x "$(command -v wine)" ]; then
 elif [ -x "$(command -v wine64)" ]; then
   WINE=wine64
 else
-  echo 'Error: bdsx requires wine. Please install wine first' >&2
-  exit $?
+  echo "try a flatpak wine"
+  WINE="flatpak run org.winehq.Wine"
+  #echo 'Error: bdsx requires wine. Please install wine first' >&2
+  #exit $?
+fi
+
+if [[ $1 == "proton" ]]; then
+  WINE="flatpak run com.valvesoftware.Steam.CompatibilityTool.Proton"
 fi
 
 # enable linux executables on wine
-- 
2.42.1

