From d3b7efd81d099cabb5626532852bef835b19a116 Mon Sep 17 00:00:00 2001
From: MisileLab <misileminecord@gmail.com>
Date: Sun, 5 Nov 2023 23:25:54 +0900
Subject: [PATCH] a furnace recipe - overcooker

---
 scripts/overcooker/main.ts | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/scripts/overcooker/main.ts b/scripts/overcooker/main.ts
index 6dcd1a0d..287f33ec 100644
--- a/scripts/overcooker/main.ts
+++ b/scripts/overcooker/main.ts
@@ -2,6 +2,13 @@ import { GameType } from "bdsx/bds/player";
 import { MobEffectIds, MobEffectInstance } from "../../bdsx/bds/effects";
 import { CANCEL } from "../../bdsx/common";
 import { events } from "../../bdsx/event";
+import { ItemStack } from "bdsx/bds/inventory";
+
+const Recipes = {
+    "furnace": {
+        "minecraft:chicken": "minecraft:cooked_chicken"
+    }
+}
 
 events.playerJoin.on((e)=>{
     // this is debug
@@ -14,7 +21,13 @@ events.blockInteractedWith.on((e)=>{
     }
     const block = e.player.getRegion().getBlock(e.blockPos);
     if (block.blockLegacy.getCommandName() == "minecraft:furnace") {
-
+        let item = e.player.getInventory().getContainer().getSlots().get(0);
+        let name = item.getName();
+        if (item == null || Recipes.furnace[name] == undefined) {
+            return CANCEL;
+        }
+        e.player.getInventory().getContainer().getItem(0).load(ItemStack.constructWith(Recipes.furnace[name], 1).save())
+        e.player.sendInventory()
     } else {
         return CANCEL;
     }
-- 
2.42.0.windows.2

